### PR TITLE
Disables OptProfv2 CI tasks if ShouldSkipOptimize = true

### PR DESCRIPTION
**Pull request by dominoFire**
## Bug

Fixes: https://github.com/NuGet/Home/issues/8578
Regression: Not a product regression
* Last working version: 5.3.0.6234
* How are we preventing it in future: More code review

## Fix

Add a condition to all but the first OptProfV2 tasks. The first task already reads the variable and skips execution if `ShouldSkipOptimize = 'true'`

## Testing/Validation

Tests Added: No
Reason for not adding tests: CI infrastructure. Hard to test in 
Validation:  CI Build on Official Build. See https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3043046&view=logs&j=8460ae35-dae7-5d3d-b89d-5bc71d24ad43

**NOTE**: if approved/merged, this commit should also be cherry-picked to `dev`.

### CODE REVIEW

**Review by zivkan — APPROVED** [review]
Please use the dev- prefix for your branch name when creating PRs, even when merging into a release-* branch.

**Comment by nkolev92** [comment]
The reason why you can squash and merge despite the failures is likely the fact that this is running on the official build definition instead of the private one.

File: build/vsts_build.yaml [hunk 1]
```diff
@@ -362,37 +362,39 @@ phases:
...
-    condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'))"
+    condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'), eq(variables['ShouldSkipOptimize'], 'false'))"
 
   - task: PowerShell@1
     displayName: 'OptProfV2:  set the TestDrop environment variable'
     inputs:
       scriptType: 'inlineScript'
       inlineScript: |
         [string] $bootstrapperInfoFilePath = "$Env:BUILD_STAGINGDIRECTORY\MicroBuild\Output\BootstrapperInfo.json"
+        Write-Host "Path: $bootstrapperInfoFilePath"
```
Comment (nkolev92) [inline build/vsts_build.yaml]: Why do you need this?

**Review by nkolev92 — COMMENTED** [review]
Looks good, I just have 1 question about the change.

File: build/vsts_build.yaml (hunk 1 above)
Comment (dominoFire) [reply build/vsts_build.yaml]: I did for diagnostig purposes, trying to figure out why we don't have a build drop in the boostraperInfo.json

Comment (dominoFire) [reply build/vsts_build.yaml]: Will remove this change

**Comment by dominoFire** [comment]
> Please use the dev- prefix for your branch name when creating PRs, even when merging into a release-* branch.

@zivkan I did it that way so that I can trigger an official build, because I needed to test in the Official CI pipeline

> The reason why you can squash and merge despite the failures is likely the fact that this is running on the official build definition instead of the private one.

@nkolev92  We should consider protecting the release branches

**Comment by nkolev92** [comment]
> @nkolev92 We should consider protecting the release branches
Yeah good point @dominoFire 

@rrelyea can do that. 

But considering our current workflow (cherry-pick into release), I don't know if that's feasible cause it might block pushing from the commandline and we certainly don't want to PR cherry-picks, unless we want to change our worfklow completely.

**Pull request merged**